File: fast_mm_python/src/control/StartAlphaFromAnother.py
"""
Start alpha optimization from a nearby solution.
"""


def StartAlphaFromAnother(q_, experiment_name, datafile_path):
    """
    Start alpha optimization from a nearby solution.
    
    Args:
        q_: q parameter value
        experiment_name: Name of the experiment
        datafile_path: Path to the data file with initial parameters
    """
    from .ExpInfo import set_q
    from .InitExp import InitExp
    from ..evaluation import Workspace, set_workspace
    from ..io import LoadParam
    
    set_q(q_)
    InitExp(experiment_name, 'alpha', True, None)
    
    workspace = Workspace()
    workspace.Build(max_level=3)
    set_workspace(workspace)
    
    LoadParam(datafile_path, None, workspace.param_manager, workspace, None)
    
    print(f"[INFO] Starting alpha optimization from provided solution.")
    print("[INFO] This is a simplified implementation - full SNOPT optimization not available.")

File: fast_mm_matlab/src/control/StartAlphaFromAnother.m
% Start from parameters 'xxx.mat' and run the climber controller for objective 'alpha'.

function StartAlphaFromAnother(q_, exp_name, param_path)
  SnInitAlpha(q_, exp_name);  % Build the environment
  LoadParam(param_path);  % Load the starting point
  global expinfo param_manager workspace;
  SnContinue();
  KeepClimb();
end

%{
Use example:
  AddPath;
  experiment_name = 'Alpha';
  StartAlphaFromAnother(5, experiment_name, 'data/K33_2.00009991.mat');
%}
